checks/evaluation/pinned_dependencies_test.go

```diff
@@ -354,3 +384,173 @@ func Test_maxScore(t *testing.T) {
+				t.Errorf("generateOwnerToDisplay() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_addWorkflowPinnedResult(t *testing.T) {
+	t.Parallel()
+	type args struct {
+		w        *worklowPinningResult
+		to       bool
+		isGitHub bool
+	}
+	tests := []struct { //nolint:govet
+		name string
+		want pinnedResult
+		args args
+	}{
+		{
+			name: "sets pinned to true if to is true",
+			args: args{
+				w:        &worklowPinningResult{},
+				to:       true,
+				isGitHub: true,
+			},
+			want: pinned,
+		},
+		{
+			name: "sets pinned to false if to is false",
+			args: args{
+				w:        &worklowPinningResult{},
+				to:       false,
+				isGitHub: true,
+			},
+			want: notPinned,
+		},
+		{
+			name: "sets pinned to undefined if to is false and isGitHub is false",
+			args: args{
+				w:        &worklowPinningResult{},
+				to:       false,
+				isGitHub: false,
+			},
+			want: pinnedUndefined,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			addWorkflowPinnedResult(tt.args.w, tt.args.to, tt.args.isGitHub)
+			if tt.args.w.gitHubOwned != tt.want {
+				t.Errorf("addWorkflowPinnedResult() = %v, want %v", tt.args.w.gitHubOwned, tt.want)
+			}
+		})
+	}
+}
+
+func TestGenerateText(t *testing.T) {
+	tests := []struct {
+		name         string
+		dependency   *checker.Dependency
+		expectedText string
+	}{
+		{
+			name: "GitHub action not pinned by hash",
+			dependency: &checker.Dependency{
+				Type: checker.DependencyUseTypeGHAction,
+				Location: &checker.File{
+					Snippet: "actions/checkout@v2",
+				},
+			},
+			expectedText: "GitHub-owned GitHubAction not pinned by hash",
+		},
+		{
+			name: "Third-party action not pinned by hash",
+			dependency: &checker.Dependency{
+				Type: checker.DependencyUseTypeGHAction,
+				Location: &checker.File{
+					Snippet: "third-party/action@v1",
+				},
+			},
+			expectedText: "third-party GitHubAction not pinned by hash",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			result := generateText(tc.dependency)
+			if !cmp.Equal(tc.expectedText, result) {
+				t.Errorf("generateText mismatch (-want +got):\n%s", cmp.Diff(tc.expectedText, result))
+			}
+		})
+	}
+}
+
+func TestUpdatePinningResults(t *testing.T) {
+	t.Parallel()
+	tests := []struct { //nolint:govet
+		name                  string
+		dependency            *checker.Dependency
+		expectedPinningResult *worklowPinningResult
+		expectedPinnedResult  map[checker.DependencyUseType]pinnedResult
+	}{
+		{
+			name: "GitHub Action - GitHub-owned",
+			dependency: &checker.Dependency{
+				Type: checker.DependencyUseTypeGHAction,
+				Location: &checker.File{
+					Snippet: "actions/checkout@v2",
+				},
+			},
+			expectedPinningResult: &worklowPinningResult{
+				thirdParties: 0,
+				gitHubOwned:  2,
+			},
+			expectedPinnedResult: map[checker.DependencyUseType]pinnedResult{},
+		},
+		{
+			name: "GitHub Action - GitHub-owned",
```

Comment:
typo here. i assume the name should reflect third party

Reply:
👍